src/libsyntax/parse/parser.rs

```diff
@@ -4686,13 +4684,13 @@ impl<'a> Parser<'a> {
             }
             (Err(mut init_err), Some((snapshot, _, ty_err))) => {  // init error, ty error
                 init_err.cancel();
-                // Couldn't parse the type nor the initializer, only raise the type error and
+                // Couldn't parse the type or the initializer; only raise the type error and
```

Comment:
"neither the type nor the initializer"?

Reply:
Yeah, but this is equally clear I think, and more common.

Reply:
It's clear, but isn't the meaning different now? Previously the comment said that both the type and the initializer couldn't be parsed. The new comment says that one of them couldn't be parsed.

Reply:
@nikic Yeah, this is the issue with parsing "or" in English – you have to know intuitively where the bracketing is, which isn't always easy, especially for non-native speakers... What it means here is: "Could not parse (the type or the initializer)". Alternatively, one can write: "Could neither parse the type nor the initializer". It's sort of an application of DeMorgan's laws, although with added fuzziness of natural language.